pandas/tests/indexes/common.py

```diff
@@ -635,6 +635,16 @@ def test_delete_base(self):
                 # either depending on numpy version
                 result = idx.delete(len(idx))
 
+    def test_equals(self):
+
+        for name, idx in compat.iteritems(self.indices):
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.astype(object)))
+            self.assertFalse(idx.equals(list(idx)))
```

Comment:
> it must be an Index for equal to compare.

@jreback that is not the current behaviour (at least for numeric indexes, see example above)